nixos/modules/system/boot/loader/limine/limine-install.py

```diff
@@ -0,0 +1,303 @@
+#!@python3@/bin/python3 -B
+
+import hashlib
+import json
+import os
+import psutil
+import re
+import shutil
+import subprocess
+import sys
+import textwrap
+
+class DotDict(dict):
+  __getattr__ = dict.__getitem__
+  __setattr__ = dict.__setitem__
+  __delattr__ = dict.__delitem__
+
+  def __init__(self, dct):
+    for key, value in dct.items():
+      if hasattr(value, 'keys'):
+        value = DotDict(value)
+      self[key] = value
```

Comment:
This class seems CC-BY-SA licensed and the script as a whole could be a deritvative, but this repo requires MIT licensing.

https://stackoverflow.com/a/13520518

Reply:
Oh I missed that, that code was not meant to make it into this version, whoops. What should I do about it? Is a comment pointing to that answer enough? I'm really not that good when it comes to licensing stuff.

Reply:
CC-BY-SA requires to redistribution under the same license. The licensing of this repository is MIT, see https://github.com/NixOS/nixpkgs/blob/master/COPYING. I would say that is incompatible.

Reply:
You can't take a CC BY-SA work and distribute it in the MIT license, but we do bundle lots of non-MIT code in Nixpkgs, and I don't think there's anything stopping anybody mixing the two, as long as they can follow the terms of both licenses.   All the patches, for example (we have a note about this in the README), and occasionally we'll just include replacement versions of files not as patches (which technically the README note doesn't cover).

So it's complicated.  What we really need in Nixpkgs is per-file license tracking, rather than the misleading idea that "nixpkgs is MIT" or whatever, which isn't really true.  Are we okay with a CC BY-SA program included in Nixpkgs?  I don't see why we wouldn't be, since we already include so many GPL etc files, which have similar conditions.  But we should be clear about it, and probably revise the note in the README to say that in addition to the thing about patches, there are other exceptions which are clearly noted.

Reply:
(But if you don't want to open that can of worms it might be easier to just not include this code. :))